app/ViewModels/CoaCaseSearchViewModel.cs

```diff
@@ -74,29 +67,25 @@ public bool Step3Complete
                 //true value means the hearing is an appeals hearing
                 if (IsAppealHearing is true)
                 {
-                    var factumFiled = FactumFiled ?? false;
-
-                    if (!factumFiled)
+                    if (FactumFiled is not true)
                     {
                         return false;
                     }
 
-                    if (IsFullDay is null)
+                    if (!IsFullDay.HasValue)
                     {
                         return false;
                     }
                 }
                 //false means chambers hearing
                 else
                 {
-                    if (SelectedApplicationTypes != null)
+                    if (SelectedApplicationTypes == null)
```

Comment:
Are `is null`, `== null` and `! .HasValue` effectively the same in this case (when we have nullable booleans)? Or what's the difference / when would you use one over the other?

Reply:
SelectedApplicationTypes is an object, not a nullable boolean so I'm checking to see if the object is null.  

`nullable is null` and `!nullable.HasValue` and `nullable .Value == null` are all essentially the same, but I think I usually favour the option that doesn't include a "!/not" operator because it's usually easier to read